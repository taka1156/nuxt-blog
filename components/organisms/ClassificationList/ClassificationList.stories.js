import ClassificationList from './ClassificationList.vue';
import {
  dummyClassificationTag,
  dummyClassificationCategory
} from '@/__testdata__/testdata.js';

export default {
  title: 'Organisms/ClassificationList',
  component: ClassificationList
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { ClassificationList },
  template: '<classification-list v-bind="$props" />'
});

export const Categories = Template.bind({});
Categories.args = {
  routePath: 'category',
  items: dummyClassificationCategory
};

export const Tags = Template.bind({});
Tags.args = { routePath: 'tag', items: dummyClassificationTag };
