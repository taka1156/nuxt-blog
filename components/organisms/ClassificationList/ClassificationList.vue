<template>
  <div>
    <div v-if="items.length !== 0">
      <ul>
        <li v-for="(item, index) in items" :key="`classification_${index}`">
          <classification-list-item
            :classification-item="item"
            :route-path="routePath"
          />
        </li>
      </ul>
    </div>
    <p v-else>{{ routePath }}がありません。</p>
  </div>
</template>

<script>
import ClassificationListItem from '../ClassificationListItem/ClassificationListItem';

export default {
  name: 'ClassificatetionList',
  components: {
    'classification-list-item': ClassificationListItem
  },
  props: {
    /**
     * タグもしくは、カテゴリーの一覧が入る
     */
    items: {
      type: Array,
      default: () => [],
      required: true
    },
    /**
     * 現在閲覧中のroute
     * (閲覧しているパスによってpageJumpの行き先が変わる)
     * @values tag, category
     */
    routePath: {
      type: String,
      default: '',
      required: true,
      validator: function(value) {
        return ['tag', 'category'].indexOf(value) !== -1;
      }
    }
  }
};
</script>

<style scoped>
/* css reset */
ul,
li {
  padding: 0;
  margin: 0;
  list-style-type: none;
}

/* css reset */
</style>
