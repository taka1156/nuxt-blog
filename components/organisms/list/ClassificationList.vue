<template>
  <div>
    <div v-if="items.length !== 0">
      <ul>
        <li v-for="(item, index) in items" :key="index">
          <classification-list-item :classification-item="item" :path="path" />
        </li>
      </ul>
    </div>
    <p v-else>{{ path }}がありません。</p>
  </div>
</template>

<script>
import ClassificationListItem from './ClassificationListItem';

export default {
  name: 'ClassificatetionList',
  components: {
    'classification-list-item': ClassificationListItem
  },
  props: {
    items: {
      type: Array,
      default: () => [],
      required: true
    },
    path: {
      type: String,
      default: '',
      required: true
    }
  }
};
</script>

<style scoped>
/* css reset */
ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

/* css reset */
</style>
