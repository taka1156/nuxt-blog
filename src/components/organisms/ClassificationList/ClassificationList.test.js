import { shallowMount } from '@vue/test-utils';
import ClassificationList from './ClassificationList.vue';
import {
  dummyClassificationCategory,
  dummyClassificationTag
} from '__testdata__/testdata.js';

describe('ClassificationList', () => {
  const classificationList = propsData =>
    shallowMount(ClassificationList, {
      propsData: {
        ...propsData
      }
    });

  it('ClassificationList初期値(Cateogry): items, routePath', () => {
    const wrapper = classificationList(dummyClassificationCategory);
    // items
    expect(wrapper.vm.$options.props.items.required).toBe(true);
    expect(wrapper.vm.items).toBe(dummyClassificationCategory.items);
    // routePath
    expect(wrapper.vm.$options.props.routePath.required).toBe(true);
    expect(wrapper.vm.routePath).toBe(dummyClassificationCategory.routePath);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('ClassificationList初期値(Tag): items, routePath', () => {
    const wrapper = classificationList(dummyClassificationTag);
    // items
    expect(wrapper.vm.$options.props.items.required).toBe(true);
    expect(wrapper.vm.items).toBe(dummyClassificationTag.items);
    // routePath
    expect(wrapper.vm.$options.props.routePath.required).toBe(true);
    expect(wrapper.vm.routePath).toBe(dummyClassificationTag.routePath);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });
});
