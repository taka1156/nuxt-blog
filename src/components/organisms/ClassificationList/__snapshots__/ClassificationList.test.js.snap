// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassificationList ClassificationList初期値(Cateogry): items, routePath 1`] = `
"<div>
  <div>
    <ul>
      <li>
        <classification-list-item-stub routepath=\\"category\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
      <li>
        <classification-list-item-stub routepath=\\"category\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
      <li>
        <classification-list-item-stub routepath=\\"category\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
      <li>
        <classification-list-item-stub routepath=\\"category\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
      <li>
        <classification-list-item-stub routepath=\\"category\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`ClassificationList ClassificationList初期値(Tag): items, routePath 1`] = `
"<div>
  <div>
    <ul>
      <li>
        <classification-list-item-stub routepath=\\"tag\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
      <li>
        <classification-list-item-stub routepath=\\"tag\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
      <li>
        <classification-list-item-stub routepath=\\"tag\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
      <li>
        <classification-list-item-stub routepath=\\"tag\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
      <li>
        <classification-list-item-stub routepath=\\"tag\\" item=\\"[object Object]\\"></classification-list-item-stub>
      </li>
    </ul>
  </div>
</div>"
`;
